### PR TITLE
fix:[CDS-68606]: Add NO_LIMIT to skip global limit configuration (ResourceLookupSyncHandler/ResourceLookupServiceImpl)

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
## Describe your changes
- This query already uses HIterator and is optimized, I just added the NO_LIMIT to it.

## Comment Triggers
<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47447)

### CODE REVIEW

**Comment by Fernando-Dourado** [comment]
trigger uts

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47447)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/main/java/software/wings/service/impl/ResourceLookupServiceImpl.java [hunk 1]
```diff
@@ -178,7 +179,7 @@ public Map<String, ResourceLookup> getResourceLookupMapWithResourceIds(String ac
     FindOptions findOptions = new FindOptions();
     findOptions.hint(BasicDBUtils.getIndexObject(ResourceLookup.mongoIndexes(), "resourceIdResourceLookupIndex"));
 
-    try (HIterator<ResourceLookup> iterator = new HIterator<>(query.fetch(findOptions))) {
+    try (HIterator<ResourceLookup> iterator = new HIterator<>(query.limit(NO_LIMIT).fetch(findOptions))) {
       while (iterator.hasNext()) {
         ResourceLookup resourceLookup = iterator.next();
         resourceLookupMap.put(resourceLookup.getResourceId(), resourceLookup);
```
Comment (bhavya181) [inline 400-rest/src/main/java/software/wings/service/impl/ResourceLookupServiceImpl.java:185]: its accumulating all resourceLookups, are you sure its not big object?

Comment (Fernando-Dourado) [reply 400-rest/src/main/java/software/wings/service/impl/ResourceLookupServiceImpl.java:185]: No, it's not a big object and it's executed a few times.

Screenshot 2023-05-08 at 11 20 47

**Review by lasdox — CHANGES_REQUESTED** [review]
This query is still not optimized, if you check underlying logic, it just uses 2 fields:
tags and resourceId (please confirm)
we need to add projections here for both these fields

**Comment by Fernando-Dourado** [comment]
Thank you @lasdox and @bhavya181 to show me changes to make it better.

**Review by bhavya181 — APPROVED** [review]

**Comment by Fernando-Dourado** [comment]
trigger review

**Pull request merged**